### DISCUSSION TITLE
CalendarDelegate.didSelectEvent(_:type:frame:): frame is relative to what view?

### DISCUSSION THREAD

**Discussion by nefch** [kvyatkovskys/KVKCalendar#280] [Q&A]
CalendarDelegate.didSelectEvent(_:type:**frame**:)

What is the coordinate system for `frame`?
`frame` is relative to what view?

Please explain! Thanks.

**Comment by kvyatkovskys** [comment]
Hello! The frame is relative of the event view.

**Reply by nefch** [reply]
OK thanks. But how do I get from the CalendarView object to this "event view"?

**Reply by kvyatkovskys** [reply]
Can you write a detailed workflow? 🤔

**Reply by nefch** [reply]
`didSelectEvent(_ event: Event, type: CalendarType, frame: CGRect?) {`
gets called and I would like to show a context menu that appears right over the rectangle that represents `event`.

It first assumed that `frame` is relative to the `CalendarView`. But it's not. This code will not work correctly:
```
alertController.popoverPresentationController?.sourceView = calendarView
alertController.popoverPresentationController?.sourceRect = frame
```
(`calendarView` is of type `CalendarView`.)

I guess I need to call either
`convert(_ rect: CGRect, to view: UIView?) -> CGRect`
or
`convert(_ rect: CGRect, from view: UIView?) -> CGRect`
but I don't know what `view` is supposed to be for these calls.

Thanks!

**Reply by kvyatkovskys** [reply]
🤔 interesting
`frame` is relative to `timelineView`. Path -> `CalendarView-DayView-TimelinePageVC-TimelineView`

**Reply by nefch** [reply]
CalendarView.dayView is **private** :-(

As a user of CalendarView I can't access the DayView class and thus `frame` is rather useless. Or am I missing something?

**Reply by nefch** [reply]
BTW `CalendarView-DayView-TimelinePageView-TimelineView` is for day view; what about week and month view?

I think there should be a library function that maps `frame` to `CalendarView`. Or `didSelectEvent` gets passed a `frame` that is already mapped to `CalendarView`.

**Reply by kvyatkovskys** [reply]
sorry for the long answer 😬 feel free to create PR with a new changes
